src/fsharp/CompileOps.fs

```diff
@@ -4889,12 +4912,42 @@ module private ScriptPreprocessClosure =
             let tcConfigB = tcConfig.CloneOfOriginalBuilder 
             TcConfig.Create(tcConfigB, validate=false),nowarns
     
-    let FindClosureFiles(closureSources, tcConfig:TcConfig, codeContext, lexResourceManager:Lexhelp.LexResourceManager) =
-        let tcConfig = ref tcConfig
+    let FindClosureFiles(mainFile, m, closureSources, origTcConfig:TcConfig, codeContext, lexResourceManager:Lexhelp.LexResourceManager) =
+        let tcConfig = ref origTcConfig
         
         let observedSources = Observed()
-        let rec loop (ClosureSource(filename,m,source,parseRequired)) = 
-          [     if not (observedSources.HaveSeen(filename)) then
+
+        // Resolve the packages
+        let rec resolvePackageManagerSources() =
+            if tcConfig.Value.packageManagerTextLines = origTcConfig.packageManagerTextLines then
+                []
+            else 
+                let referenceLoadingResult =
+                    ReferenceLoading.PaketHandler.Internals.ResolvePackages
+                        Microsoft.FSharp.Compiler.ReferenceLoading.PaketHandler.targetFramework
+                        Microsoft.FSharp.Compiler.ReferenceLoading.PaketHandler.GetCommandForTargetFramework
+                        (fun workDir -> Microsoft.FSharp.Compiler.ReferenceLoading.PaketHandler.GetPaketLoadScriptLocation workDir (Some Microsoft.FSharp.Compiler.ReferenceLoading.PaketHandler.targetFramework))
+                        Microsoft.FSharp.Compiler.ReferenceLoading.PaketHandler.AlterPackageManagementToolCommand
+                        (tcConfig.Value.implicitIncludeDir, mainFile, tcConfig.Value.packageManagerTextLines)
+                match referenceLoadingResult with 
+                | ReferenceLoading.PaketHandler.ReferenceLoadingResult.PackageManagerNotFound (implicitIncludeDir, userProfile) ->
+                    errorR(Error(FSComp.SR.packageManagerNotFound(implicitIncludeDir, userProfile),m))
```

Comment:
You'll only get a red squiggly error in the VS UI if the ``m`` is a correct range in the current file being processed